internal/adapter/mssqldb/todo_repo.go

```diff
@@ -0,0 +1,52 @@
+package mssqldb
+
+import (
+	"todoapp/internal/entity"
+	"todoapp/internal/usecase/repo"
+
+	"gorm.io/gorm"
+)
+
+// check if missing implementation
+var _ repo.TodoRepo = new(todoRepo)
+
+func NewTodoRepo(db *gorm.DB) repo.TodoRepo {
+	return &todoRepo{db}
+}
+
+type todoRepo struct {
+	db *gorm.DB
+}
+
+func (tr *todoRepo) GetTodoListOf(username string) []entity.Todo {
+	var todos []entity.Todo
+	tr.db.Where("username = ?", username).Find(&todos)
+	return todos
+}
+
+func (tr *todoRepo) GetTodoByID(id string) *entity.Todo {
+	var todo entity.Todo
+	tr.db.Where("id = ?", id).Find(&todo)
+	return &todo
+}
+
+func (tr *todoRepo) CreateTodo(todo *entity.Todo) error {
+	result := tr.db.Create(todo)
+	if result.Error != nil {
+		return result.Error
```

Comment:
change to `return errors.New("cannot create todo using gorm: " + result.Error.Error())`